Bitkit/Services/BackupService.swift

```diff
@@ -381,8 +482,95 @@ class BackupService {
                 return widgetsData
             }
 
-        case .wallet, .metadata, .blocktank, .slashtags, .ldkActivity, .lightningConnections:
-            throw NSError(domain: "BackupService", code: -1, userInfo: [NSLocalizedDescriptionKey: "\(category.rawValue) backup not yet implemented"])
+        case .wallet:
+            let transfers = try TransferStorage.shared.getAll()
+            let payload = WalletBackupV1(
+                version: 1,
+                createdAt: UInt64(Date().timeIntervalSince1970),
+                transfers: transfers
+            )
+            return try JSONEncoder().encode(payload)
+
+        case .metadata:
+            let currentTime = UInt64(Date().timeIntervalSince1970)
+
+            var tagMetadata: [TagMetadataItem] = []
+            do {
+                let activities = try await CoreService.shared.activity.get()
+                for activity in activities {
+                    let activityId = switch activity {
+                    case let .lightning(ln): ln.id
+                    case let .onchain(on): on.id
+                    }
+
+                    let tags = try await CoreService.shared.activity.tags(forActivity: activityId)
```

Comment:
aside:
I suppose iOS doesn't support tags for receive transactions?!

Not sure it's doable without storing `tagMetadata` separately, and then attaching it to the activity item when the payment for that bip21 is received. 🤔